### ISSUE TITLE
🐛 Bug Report: TypeError when trying to override `BackstageSidebarSubmenu` in custom theme

### ISSUE THREAD

**Issue by awanlin** [backstage/backstage#25615]
### 📜 Description

We get a TypeError when trying to override `BackstageSidebarSubmenu` in custom theme

### 👍 Expected behavior

If should just work like any of the other everrides

### 👎 Actual Behavior with Screenshots

We get the following TypeError:

```text
Object literal may only specify known properties, but 'BackstageSidebarSubmenu' does not exist in type 'Components<Omit<Theme, "components">>'.  Did you mean to write 'BackstageSidebar'?
ts(2561)index.d.ts(201, 5): The expected type comes from property 'components' which is declared here on type 'UnifiedThemeOptions'
```

### 👟 Reproduction steps

1. Create a new Backstage instance
2. Create a custom theme following the docs: https://backstage.io/docs/getting-started/app-custom-theme
3. Add an override for the `BackstageSidebarSubmenu` like this:

```ts
BackstageSidebarSubmenu: {
      styleOverrides: {
        title: ({ theme }) => ({
          color: theme.page.fontColor,
          fontWeight: 900,
        }),
      },
    },
```

**Notice:** you get a TypeError

### 📃 Provide the context for the Bug.

Want to be able to use this override to make our custom theme look great!

### 🖥️ Your Environment

I used the Demo site to prove this out:

```text
OS:   Darwin 23.5.0 - darwin/arm64
node: v18.18.2
yarn: 3.5.0
cli:  0.26.11-next.1 (installed)
backstage:  1.29.0-next.2

Dependencies:
  @backstage/app-defaults                                          1.5.8-next.2
  @backstage/backend-app-api                                       0.7.8, 0.7.10-next.1
  @backstage/backend-common                                        0.21.7, 0.23.1, 0.23.3-next.1
  @backstage/backend-defaults                                      0.3.4-next.1
  @backstage/backend-dev-utils                                     0.1.4
  @backstage/backend-openapi-utils                                 0.1.13, 0.1.15-next.1
  @backstage/backend-plugin-api                                    0.6.20, 0.6.22-next.1
  @backstage/backend-tasks                                         0.5.25, 0.5.27-next.1
  @backstage/catalog-client                                        1.6.5
  @backstage/catalog-model                                         1.5.0
  @backstage/cli-common                                            0.1.14
  @backstage/cli-node                                              0.2.6
  @backstage/cli                                                   0.26.11-next.1
  @backstage/config-loader                                         1.8.1
  @backstage/config                                                1.2.0
  @backstage/core-app-api                                          1.13.1-next.1
  @backstage/core-compat-api                                       0.2.6, 0.2.7-next.1
  @backstage/core-components                                       0.14.8, 0.14.9-next.1
  @backstage/core-plugin-api                                       1.9.3
  @backstage/e2e-test-utils                                        0.1.1
  @backstage/errors                                                1.2.4
  @backstage/eslint-plugin                                         0.1.8
  @backstage/frontend-plugin-api                                   0.6.6, 0.6.7-next.1
  @backstage/integration-aws-node                                  0.1.12
  @backstage/integration-react                                     1.1.28, 1.1.29-next.0
  @backstage/integration                                           1.12.0, 1.13.0-next.0
  @backstage/plugin-api-docs                                       0.11.7-next.2
  @backstage/plugin-app-backend                                    0.3.71-next.1
  @backstage/plugin-app-node                                       0.1.22-next.1
  @backstage/plugin-auth-backend-module-atlassian-provider         0.2.3-next.1
  @backstage/plugin-auth-backend-module-aws-alb-provider           0.1.14-next.1
  @backstage/plugin-auth-backend-module-azure-easyauth-provider    0.1.5-next.1
  @backstage/plugin-auth-backend-module-bitbucket-provider         0.1.5-next.1
  @backstage/plugin-auth-backend-module-cloudflare-access-provider 0.1.5-next.1
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.17-next.1
  @backstage/plugin-auth-backend-module-github-provider            0.1.19-next.1
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.19-next.1
  @backstage/plugin-auth-backend-module-google-provider            0.1.19-next.1
  @backstage/plugin-auth-backend-module-guest-provider             0.1.8-next.1
  @backstage/plugin-auth-backend-module-microsoft-provider         0.1.17-next.1
  @backstage/plugin-auth-backend-module-oauth2-provider            0.2.3-next.1
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.1.15-next.1
  @backstage/plugin-auth-backend-module-oidc-provider              0.2.3-next.1
  @backstage/plugin-auth-backend-module-okta-provider              0.0.15-next.1
  @backstage/plugin-auth-backend-module-onelogin-provider          0.1.3-next.1
  @backstage/plugin-auth-backend                                   0.22.9-next.1
  @backstage/plugin-auth-node                                      0.4.15, 0.4.17-next.1
  @backstage/plugin-auth-react                                     0.1.4-next.1
  @backstage/plugin-bitbucket-cloud-common                         0.2.21-next.0
  @backstage/plugin-catalog-backend-module-logs                    0.0.1-next.1
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.20-next.1
  @backstage/plugin-catalog-backend                                1.24.0-next.1
  @backstage/plugin-catalog-common                                 1.0.24
  @backstage/plugin-catalog-graph                                  0.4.7-next.2
  @backstage/plugin-catalog-node                                   1.12.2, 1.12.4-next.1
  @backstage/plugin-catalog-react                                  1.12.1, 1.12.2-next.2
  @backstage/plugin-catalog                                        1.21.1-next.2
  @backstage/plugin-events-node                                    0.3.8-next.1
  @backstage/plugin-home-react                                     0.1.15-next.2
  @backstage/plugin-home                                           0.7.7-next.2
  @backstage/plugin-kubernetes-backend                             0.18.3-next.1
  @backstage/plugin-kubernetes-common                              0.8.0
  @backstage/plugin-kubernetes-node                                0.1.16-next.1
  @backstage/plugin-kubernetes-react                               0.4.1-next.1
  @backstage/plugin-kubernetes                                     0.11.12-next.2
  @backstage/plugin-org                                            0.6.27-next.2
  @backstage/plugin-permission-common                              0.7.14
  @backstage/plugin-permission-node                                0.7.31, 0.7.33-next.1
  @backstage/plugin-permission-react                               0.4.23
  @backstage/plugin-proxy-backend                                  0.5.3-next.1
  @backstage/plugin-scaffolder-backend-module-azure                0.1.14-next.1
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud      0.1.12-next.1
  @backstage/plugin-scaffolder-backend-module-bitbucket-server     0.1.12-next.1
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.2.12-next.1
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.14-next.1
  @backstage/plugin-scaffolder-backend-module-gitea                0.1.12-next.1
  @backstage/plugin-scaffolder-backend-module-github               0.4.0-next.1
  @backstage/plugin-scaffolder-backend-module-gitlab               0.4.4-next.1
  @backstage/plugin-scaffolder-backend                             1.23.0-next.2
  @backstage/plugin-scaffolder-common                              1.5.3
  @backstage/plugin-scaffolder-node                                0.4.8-next.1
  @backstage/plugin-scaffolder-react                               1.10.0-next.2
  @backstage/plugin-scaffolder                                     1.22.1-next.2
  @backstage/plugin-search-backend-module-catalog                  0.1.28-next.1
  @backstage/plugin-search-backend-module-explore                  0.1.26, 0.1.28-next.1
  @backstage/plugin-search-backend-module-techdocs                 0.1.27-next.1
  @backstage/plugin-search-backend-node                            1.2.25, 1.2.27-next.1
  @backstage/plugin-search-backend                                 1.5.14-next.1
  @backstage/plugin-search-common                                  1.2.12
  @backstage/plugin-search-react                                   1.7.12, 1.7.13-next.1
  @backstage/plugin-search                                         1.4.14-next.2
  @backstage/plugin-signals-react                                  0.0.4
  @backstage/plugin-techdocs-backend                               1.10.9-next.1
  @backstage/plugin-techdocs-node                                  1.12.8-next.1
  @backstage/plugin-techdocs-react                                 1.2.6-next.1
  @backstage/plugin-techdocs                                       1.10.7-next.2
  @backstage/plugin-user-settings-common                           0.0.1
  @backstage/plugin-user-settings                                  0.8.9-next.2
  @backstage/release-manifests                                     0.0.11
  @backstage/theme                                                 0.5.6
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.8
  ```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, I don't have time to work on this right now

**Comment by awanlin** [comment]
Digging into this more it seems that if it's not in this list you can't override it:

https://github.com/backstage/backstage/blob/5442edda318637e7b40a4991bf9b60e3b412b9b5/packages/core-components/src/overridableComponents.ts#L95

I didn't think that was needed, it was never something I ever remember seeing in the [customization docs](https://backstage.io/docs/getting-started/app-custom-theme#overriding-backstage-and-material-ui-components-styles) (yes, I'm the one who's updated them a few times already) 🤔

**Comment by awanlin** [comment]
Resolved by #25618
